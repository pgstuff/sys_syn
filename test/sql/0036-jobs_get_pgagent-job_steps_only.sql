BEGIN;

CREATE EXTENSION tinyint SCHEMA public;
CREATE EXTENSION sys_syn;

CREATE SCHEMA user_data AUTHORIZATION postgres;

CREATE TABLE user_data.in_1_test_table (
        in_1_test_table_key integer NOT NULL,
        in_1_test_table_text text,
        CONSTRAINT in_1_test_table_pkey PRIMARY KEY (in_1_test_table_key));

CREATE TABLE user_data.in_2_test_table (
        in_2_test_table_key integer NOT NULL,
        in_2_test_table_text text,
        CONSTRAINT in_2_test_table_pkey PRIMARY KEY (in_2_test_table_key));

INSERT INTO sys_syn.in_groups_def VALUES ('in_1');

DO $$BEGIN
        EXECUTE sys_syn.in_table_add_sql('user_data.in_1_test_table'::regclass, 'in_1');
END$$;

INSERT INTO sys_syn.in_groups_def VALUES ('in_2');

DO $$BEGIN
        EXECUTE sys_syn.in_table_add_sql('user_data.in_2_test_table'::regclass, 'in_2');
END$$;

INSERT INTO sys_syn.out_groups_def VALUES ('out_1');

SELECT sys_syn.out_table_add('user_data', 'in_1_test_table', 'out_1');
SELECT sys_syn.out_table_add('user_data', 'in_2_test_table', 'out_1');

INSERT INTO sys_syn.out_groups_def VALUES ('out_2');

SELECT sys_syn.out_table_add('user_data', 'in_1_test_table', 'out_2');
SELECT sys_syn.out_table_add('user_data', 'in_2_test_table', 'out_2');

SELECT sys_syn.in_pull_sequence_populate_assume();

SELECT sys_syn.jobs_get_pgagent(delete_sql => true, create_sql => true, job_steps_only => true);

ROLLBACK;
