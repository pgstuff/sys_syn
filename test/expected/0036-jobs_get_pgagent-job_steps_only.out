BEGIN;
CREATE EXTENSION tinyint SCHEMA public;
CREATE EXTENSION sys_syn;
CREATE SCHEMA user_data AUTHORIZATION postgres;
CREATE TABLE user_data.in_1_test_table (
        in_1_test_table_id integer NOT NULL,
        in_1_test_table_text text,
        CONSTRAINT in_1_test_table_pid PRIMARY KEY (in_1_test_table_id));
CREATE TABLE user_data.in_2_test_table (
        in_2_test_table_id integer NOT NULL,
        in_2_test_table_text text,
        CONSTRAINT in_2_test_table_pid PRIMARY KEY (in_2_test_table_id));
INSERT INTO sys_syn.in_groups_def VALUES ('in_1');
DO $$BEGIN
        EXECUTE sys_syn.in_table_add_sql('user_data.in_1_test_table'::regclass, 'in_1');
END$$;
INSERT INTO sys_syn.in_groups_def VALUES ('in_2');
DO $$BEGIN
        EXECUTE sys_syn.in_table_add_sql('user_data.in_2_test_table'::regclass, 'in_2');
END$$;
INSERT INTO sys_syn.out_groups_def VALUES ('out_1');
SELECT sys_syn.out_table_add('user_data', 'in_1_test_table', 'out_1');
 out_table_add 
---------------
 
(1 row)

SELECT sys_syn.out_table_add('user_data', 'in_2_test_table', 'out_1');
 out_table_add 
---------------
 
(1 row)

INSERT INTO sys_syn.out_groups_def VALUES ('out_2');
SELECT sys_syn.out_table_add('user_data', 'in_1_test_table', 'out_2');
 out_table_add 
---------------
 
(1 row)

SELECT sys_syn.out_table_add('user_data', 'in_2_test_table', 'out_2');
 out_table_add 
---------------
 
(1 row)

SELECT sys_syn.in_pull_sequence_populate_assume();
 in_pull_sequence_populate_assume 
----------------------------------
 
(1 row)

SELECT sys_syn.jobs_get_pgagent(delete_sql => true, create_sql => true, job_steps_only => true);
                                                            jobs_get_pgagent                                                            
----------------------------------------------------------------------------------------------------------------------------------------
 BEGIN;                                                                                                                                +
                                                                                                                                       +
 DELETE FROM pgagent.pga_jobsteplog                                                                                                    +
 USING   pgagent.pga_jobstep                                                                                                           +
 WHERE   pga_jobsteplog.jsljstid = pga_jobstep.jstid AND                                                                               +
         (pga_jobstep.jstjobid = (SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_1') OR   +
         pga_jobstep.jstjobid = (SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Processed Sequence in_1'));+
                                                                                                                                       +
 DELETE FROM pgagent.pga_jobstep                                                                                                       +
 WHERE   jstjobid = (SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_1') OR                +
         jstjobid = (SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Processed Sequence in_1');             +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_1'),                             +
         '1-distribute_load',                                                                                                          +
         '',             true,           's',                                                                                          +
         'SELECT sys_syn.distribute_load(60)',                                                                                         +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_1'),                             +
         '3-vacuum 00001-in_1_test_table',                                                                                             +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_1_test_table_vacuum()',                                                                                  +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_1'),                             +
         '4-pull 00001-in_1_test_table',                                                                                               +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_1_test_table_pull(FALSE)',                                                                               +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_1'),                             +
         '5-move 00001-in_1_test_table 00001-out_1',                                                                                   +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_1_test_table_out_1_move()',                                                                              +
         '',             'contrib_regression',      'f');;                                                                             +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_1'),                             +
         '5-move 00001-in_1_test_table 00003-out_2',                                                                                   +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_1_test_table_out_2_move()',                                                                              +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Processed Sequence in_1'),                        +
         '1-distribute_load',                                                                                                          +
         '',             true,           's',                                                                                          +
         'SELECT sys_syn.distribute_load(60)',                                                                                         +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Processed Sequence in_1'),                        +
         '2-processed 00001-in_1_test_table 00001-in_1_test_table 00001-out_1',                                                        +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_1_test_table_out_1_processed()',                                                                         +
         '',             'contrib_regression',      'f');;                                                                             +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Processed Sequence in_1'),                        +
         '2-processed 00001-in_1_test_table 00001-in_1_test_table 00003-out_2',                                                        +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_1_test_table_out_2_processed()',                                                                         +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 COMMIT;                                                                                                                               +
 BEGIN;                                                                                                                                +
                                                                                                                                       +
 DELETE FROM pgagent.pga_jobsteplog                                                                                                    +
 USING   pgagent.pga_jobstep                                                                                                           +
 WHERE   pga_jobsteplog.jsljstid = pga_jobstep.jstid AND                                                                               +
         (pga_jobstep.jstjobid = (SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_2') OR   +
         pga_jobstep.jstjobid = (SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Processed Sequence in_2'));+
                                                                                                                                       +
 DELETE FROM pgagent.pga_jobstep                                                                                                       +
 WHERE   jstjobid = (SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_2') OR                +
         jstjobid = (SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Processed Sequence in_2');             +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_2'),                             +
         '1-distribute_load',                                                                                                          +
         '',             true,           's',                                                                                          +
         'SELECT sys_syn.distribute_load(60)',                                                                                         +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_2'),                             +
         '3-vacuum 00002-in_2_test_table',                                                                                             +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_2_test_table_vacuum()',                                                                                  +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_2'),                             +
         '4-pull 00002-in_2_test_table',                                                                                               +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_2_test_table_pull(FALSE)',                                                                               +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_2'),                             +
         '5-move 00002-in_2_test_table 00002-out_1',                                                                                   +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_2_test_table_out_1_move()',                                                                              +
         '',             'contrib_regression',      'f');;                                                                             +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Pull Sequence in_2'),                             +
         '5-move 00002-in_2_test_table 00004-out_2',                                                                                   +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_2_test_table_out_2_move()',                                                                              +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Processed Sequence in_2'),                        +
         '1-distribute_load',                                                                                                          +
         '',             true,           's',                                                                                          +
         'SELECT sys_syn.distribute_load(60)',                                                                                         +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Processed Sequence in_2'),                        +
         '2-processed 00002-in_2_test_table 00001-in_2_test_table 00002-out_1',                                                        +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_2_test_table_out_1_processed()',                                                                         +
         '',             'contrib_regression',      'f');;                                                                             +
                                                                                                                                       +
 INSERT INTO pgagent.pga_jobstep (                                                                                                     +
         jstjobid,                                                                                                                     +
         jstname,                                                                                                                      +
         jstdesc,        jstenabled,     jstkind,                                                                                      +
         jstcode,                                                                                                                      +
         jstconnstr,     jstdbname,      jstonerror)                                                                                   +
 VALUES ((SELECT pga_job.jobid FROM pgagent.pga_job WHERE pga_job.jobname = 'sys_syn Processed Sequence in_2'),                        +
         '2-processed 00002-in_2_test_table 00001-in_2_test_table 00004-out_2',                                                        +
         '',             true,           's',                                                                                          +
         'SELECT user_data.in_2_test_table_out_2_processed()',                                                                         +
         '',             'contrib_regression',      'f');                                                                              +
                                                                                                                                       +
 COMMIT;
(1 row)

ROLLBACK;
